modules/plugins/languages/scala.nix

```
@@ -0,0 +1,141 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (lib) types optionalString;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.nvim.types) mkGrammarOption;
+  inherit (lib.nvim.binds) mkMappingOption;
+  inherit (lib.options) mkOption mkEnableOption;
+  inherit (lib.nvim.dag) entryAfter;
+
+  mkBinding = binding: "vim.api.nvim_buf_set_keymap(bufnr, 'n', '${binding.value}', '<cmd>lua ${binding.action}<CR>', {noremap=true, silent=true, desc='${binding.description}'})";
+
+  listCommandsBinding = {
+    value = cfg.lsp.extraMappings.listCommands;
+    description = "List Metals commands";
+    action =
+      if config.vim.telescope.enable
+      then ''require("telescope").extensions.metals.commands()''
+      else ''require("metals").commands()'';
+  };
+
+  cfg = config.vim.languages.scala;
+
+  usingDap = config.vim.debugger.nvim-dap.enable && cfg.dap.enable;
+  usingLualine = config.vim.statusline.lualine.enable;
+in {
+  options.vim.languages.scala = {
+    enable = mkEnableOption "Scala language support";
+    treesitter = {
+      enable = mkEnableOption "Scala treesitter" // {default = config.vim.languages.enableTreesitter;};
+      package = mkGrammarOption pkgs "scala";
+    };
+    lsp = {
```

```suggestion
    };
    
    lsp = {
```

Here and in other instances. Cramped code is not very readable.